### PR TITLE
Cleanup and styles

### PR DESCRIPTION
**Pull request by weerd**
#### What's this PR do?
This PR does a little bit of cleanup and updates to docblock comments etc. It breaks out the Competitions CRUD form into a partial so it can be included in both the create & edit forms instead of having the same form repeated, along with doing some model binding like the WaitingRooms CRUD does. It also adds errors to Competitions crud and adds some styling related things for said errors.

#### Where should the reviewer start?
Mostly just review the form stuff and see if all looks right.

#### How should this be manually tested?
Meh, pull down the branch and try out the crud forms and make sure they still work.

---
@angaither @sbsmith86

### CODE REVIEW

File: app/Http/Controllers/CompetitionsController.php [hunk 1]
```diff
@@ -32,7 +32,7 @@ public function index()
     {
         $competitions = Competition::all();
 
-        return view('competitions.index')->withCompetitions($competitions);
+        return view('competitions.index', compact('competitions'));
```
Comment (angaither) [inline app/Http/Controllers/CompetitionsController.php]: RUTHLESS

**Comment by sbsmith86** [comment]
looks good to me

**Comment by angaither** [comment]
:+1:

File: app/Http/Controllers/CompetitionsController.php (hunk 1 above)
Comment (weerd) [inline app/Http/Controllers/CompetitionsController.php]: lol. :trollface: :

**Pull request merged**